src/routes/bounce_activities_page.jsx

```diff
@@ -72,73 +81,98 @@ export default class BounceActivityPage extends React.Component {
     const newQuery = {
       ...filterQuery,
       filterBy: value.toLowerCase(),
-      option: "",
+      option: ""
     };
     this.setState({
-      filterQuery: newQuery,
+      filterQuery: newQuery
     });
   }
 
-  async updateFilterOption(e) {
-    const { filterQuery } = this.state;
+  async filterActivities(value) {
+    const { filterQuery, currentPageIndex, rulesToShow } = this.state;
     const { filterBy } = filterQuery;
-    const { value } = e.target;
     const newQuery = { ...filterQuery, option: value.toLowerCase() };
-
-    const filter = { limit: 99999, offset: 1, filterBy, option: value };
+    const filter = {
+      limit: rulesToShow,
+      offset: currentPageIndex - 1,
+      filterBy,
+      option: value
+    };
     try {
-      const { data, status } = await getFilteredActivityLog(filter);
+      const { data, status } = await getActivityLog(filter);
       if (status === 200) {
         this.setState({
-          rules: data.reverse(),
+          activityLog: data.reverse(),
           numRules: data.length,
+          filterQuery: newQuery
         });
       }
     } catch (err) {
       this.setState({
-        isNetworkError: true,
+        isNetworkError: true
       });
     }
+  }
 
-    this.setState({
-      filterQuery: newQuery,
-    });
+  updateFilterOption(e) {
+    const { value } = e.target;
+    this.handleFilterActivities(value);
   }
 
-  async handleOptionSelector(e) {
-    const { filterQuery } = this.state;
-    const { filterBy } = filterQuery;
+  handleOptionSelector(e) {
     const { value } = e;
-    const newQuery = { ...filterQuery, option: value.toLowerCase() };
+    this.filterActivities(value);
+  }
 
-    const filter = { limit: 99999, offset: 1, filterBy, option: value };
+  async onDateChange(dateRange) {
+    const { focusedInput } = this.state;
+    this.setState(
+      {
+        [focusedInput]: dateRange[focusedInput]
+      },
+      () => {
+        const { startDate, endDate } = this.state;
+        if (startDate !== null && endDate !== null) {
+          this.filterActivities(
+            `${startDate.startOf("day").unix()} ${endDate.endOf("day").unix()}`
+          );
+        }
+      }
+    );
+  }
+
+  onFocusChange(focusedInput) {
+    this.setState({
+      focusedInput
+    });
+  }
+
+  async handleClearSearch() {
+    const { currentPageIndex } = this.state;
     try {
-      const { data, status } = await getFilteredActivityLog(filter);
+      const { data, status } = await getActivityLog({
+        limit: 99999,
```

Comment:
Same here so when the value of the limit changes we only need to change one constant variable at the top